test/clj_kondo/analysis_test.clj

```diff
@@ -123,17 +123,37 @@
         (assert-submaps
          '[{:name "a" :ns xml}]
          (:keywords a))))
+    ;; Don't use assertmap here to make sure ns is absent
     (testing "no namespace for key :a"
       (let [a (analyze "#:xml{:_/a 1}"
                        {:config {:output {:analysis {:keywords true}}}})]
-        (is (= '{:row 1, :col 7, :end-row 1, :end-col 11, :name "a", :filename "<stdin>"}
-               (first (:keywords a))))))
+        (is (= '[{:row 1, :col 7, :end-row 1, :end-col 11, :name "a", :filename "<stdin>" :auto-resolved false :namespace-from-prefix false}]
+               (:keywords a)))))
+    ;; Don't use assertmap here to make sure ns is absent
     (testing "no namespace for key :b"
       (let [a (analyze "#:xml{:a {:b 1}}"
                        {:config {:output {:analysis {:keywords true}}}})]
-        (is (= '[{:row 1, :col 7, :end-row 1, :end-col 9, :ns xml, :name "a", :filename "<stdin>"}
-                 {:row 1, :col 11, :end-row 1, :end-col 13, :name "b", :filename "<stdin>"}]
-               (:keywords a)))))))
+        (is (= '[{:row 1, :col 7, :end-row 1, :end-col 9, :ns xml, :name "a", :filename "<stdin>" :auto-resolved false :namespace-from-prefix true}
+                 {:row 1, :col 11, :end-row 1, :end-col 13, :name "b", :filename "<stdin>" :auto-resolved false :namespace-from-prefix false}]
+          (:keywords a)))))
+    (testing "auto-resolved"
+      (let [a (analyze "(ns foo)
+                        :a ::b :bar/c
+                        #:d{:e 1 :_/f 2 :g/h 3 ::i 4}
+                        {:j/k 5 :l 6 ::m 7}"
+                       {:config {:output {:analysis {:keywords true}}}})]
+        (assert-submaps
+              '[{:name "a" :auto-resolved false :namespace-from-prefix false}
```

Comment:
I think it would be less noisy if we didn't add the keys if the value is `false`.

Reply:
I thought about that, it works for me, but it will make test larger as well as assert-submap can't test keys absents

Reply:
Small price to pay compared to having larger payloads in the analysis output.